fearless-utils/src/main/java/jp/co/soramitsu/fearless_utils/runtime/metadata/RuntimeMetadataTopReader.kt

```diff
@@ -0,0 +1,38 @@
+package jp.co.soramitsu.fearless_utils.runtime.metadata
+
+import jp.co.soramitsu.fearless_utils.extensions.fromHex
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.RuntimeMetadataSchemaV14
+import jp.co.soramitsu.fearless_utils.scale.EncodableStruct
+import jp.co.soramitsu.fearless_utils.scale.Schema
+import jp.co.soramitsu.fearless_utils.scale.uint32
+import jp.co.soramitsu.fearless_utils.scale.uint8
+
+object Magic : Schema<Magic>() {
+    val magicNumber by uint32()
+    val runtimeVersion by uint8()
+}
+
+object RuntimeMetadataReader {
+    private var magic: EncodableStruct<*>? = null
+    private var schema: EncodableStruct<*>? = null
+    fun read(s: String): RuntimeMetadataReader {
+        val bytes = s.fromHex()
```

Comment:
Having mutable variables inside the singleton is not thread safe and may fail if several runtimes are read concurrently. I suggest having `RuntimeMetadataReader` as a class with immutable variables and a factory method which will create its instance (the same that `read` is currecntly doing)